### ISSUE TITLE
Fields inside the super table field get a new UID every time the field is saved

### ISSUE THREAD

**Issue by MoritzLost** [engram-design/SuperTable#482]
### Describe the bug

Every time I'm saving a Super Table field (either a standalone field or a field nested in a matrix field), all the nested fields inside that are part of the super table field get a new UID. This results in a lot of noise in our git commits. All the changes look like this:

```diff
diff --git a/config/project/superTableBlockTypes/9f7e34d7-23a3-4f79-805f-d77c5988ecc4.yaml b/config/project/superTableBlockTypes/9f7e34d7-23a3-4f79-805f-d77c5988ecc4.yaml
index c024d52..f97d49c 100644
--- a/config/project/superTableBlockTypes/9f7e34d7-23a3-4f79-805f-d77c5988ecc4.yaml
+++ b/config/project/superTableBlockTypes/9f7e34d7-23a3-4f79-805f-d77c5988ecc4.yaml
@@ -13,7 +13,7 @@ fieldLayouts:
             required: false
             tip: null
             type: craft\fieldlayoutelements\CustomField
-            uid: 078033d5-8bcd-46de-bf8c-ac995f6c8042
+            uid: 890ab4db-2f15-4e34-97fb-3cdee6b8752d
             userCondition: null
             warning: null
             width: 100
@@ -25,7 +25,7 @@ fieldLayouts:
             required: false
             tip: null
             type: craft\fieldlayoutelements\CustomField
-            uid: 5376d530-b3ec-4b38-9237-4af61adc7ea3
+            uid: 00588734-b48e-4523-bca5-04a82a0ac37f
             userCondition: null
             warning: null
             width: 100
@@ -37,7 +37,7 @@ fieldLayouts:
             required: false
             tip: null
             type: craft\fieldlayoutelements\CustomField
-            uid: 8302814f-c068-4c67-9b05-99f1de1e2b35
+            uid: 09a2aa31-3721-4f0b-82bf-4797ad4ded6c
             userCondition: null
             warning: null
             width: 100
@@ -49,7 +49,7 @@ fieldLayouts:
             required: false
             tip: null
             type: craft\fieldlayoutelements\CustomField
-            uid: c1c05647-e550-4f65-ba9e-8a1158a35219
+            uid: 9e4c4c2b-15c1-456c-89b7-b959fea2b4a5
             userCondition: null
             warning: null
             width: 100
```

I think this is the [same issue that existed with matrix fields](https://github.com/craftcms/cms/issues/11248) after the Craft 4 updates. In this case, maybe this could be fixed for Super Table [in the same way](https://github.com/craftcms/cms/commit/0405a4be3fb0cd303ae2dd2368991b06f9b08399)? See the linked issue for details.

### Steps to reproduce

1. Create a Super Table field and save it.
2. Use `git add .` to stage the config file for the super table field (or save the config file otherwise).
3. Save the field again.
4. Compare the config file to the previous state. All the fields inside the super table field will have a new UID.

### Craft CMS version

4.2.1.1

### Plugin version

3.0.0

### Multi-site?

No

**Comment by engram-design** [comment]
Good call! I believe I was part-way through making this change, but must have gotten distracted. Should be fixed for the next release. To get the fix early, change your `verbb/super-table` requirement in `composer.json` to:

```
"require": {
  "verbb/super-table": "dev-craft-4 as 3.0.0",
  "...": "..."
}
```

Then run `composer update`.

**Comment by MoritzLost** [comment]
@engram-design Awesome, thanks for the fix & the quick response! 🚀

**Comment by engram-design** [comment]
Fixed in [3.0.1](https://github.com/verbb/super-table/releases/tag/3.0.1)
